src/app/(nextjs_migration)/(guest)/breach-details/[breachName]/page.tsx

```diff
@@ -0,0 +1,368 @@
+      "PDL",
+    ].includes(breach.Name)
+  ) {
+    return "data-aggregator-breach";
+  }
+  if (breach.IsSensitive) {
+    return "sensitive-breach";
+  }
+  if (breach.Domain !== "") {
+    return "website-breach";
+  }
+  return "data-aggregator-breach";
+}
+
+function compareBreachDates(breach: Breach) {
+  const breachDate = new Date(breach.BreachDate);
+  const addedDate = new Date(breach.AddedDate);
+  const timeDiff = Math.abs(breachDate.getTime() - addedDate.getTime());
+  const dayDiff = Math.ceil(timeDiff / (1000 * 3600 * 24));
+  if (dayDiff > 90) {
+    return true;
+  }
+  return false;
+}
+function getSortedDataClasses(
+  breach: Breach,
+  isUserBrowserFirefox = false,
+  isUserLocaleEnUs = false,
+  isUserLocalEn = false,
+  changePWLink = false
+) {
+  const l10n = getL10n();
+  const priorityDataClasses: any = getAllPriorityDataClasses(
+    isUserBrowserFirefox,
+    isUserLocaleEnUs,
+    changePWLink
+  );
+
+  const sortedDataClasses = {
+    priority: [] as any[],
+    lowerPriority: [] as any[],
+  };
+
+  const dataClasses: any[] = breach.DataClasses;
+  dataClasses.forEach((dataClass) => {
+    const dataType = l10n.getString(dataClass);
+    if (priorityDataClasses[dataClass]) {
+      priorityDataClasses[dataClass].dataType = dataType;
+      sortedDataClasses.priority.push(priorityDataClasses[dataClass]);
+      return;
+    }
+    sortedDataClasses.lowerPriority.push(dataType);
+  });
+  sortedDataClasses.priority.sort((a, b) => {
+    return b.weight - a.weight;
+  });
+  sortedDataClasses.lowerPriority = sortedDataClasses.lowerPriority.join(
+    ", "
+  ) as any;
+  return sortedDataClasses;
+}
+
+function makeDataSection(breach: Breach) {
+  const dataClasses = getSortedDataClasses(breach);
+
+  const output = dataClasses.priority.map((dataClass, dataIndex) => (
+    <li key={`data-class-${dataClass.glyphName}`}>
+      <Image src={glyphs[dataClass.glyphName]} width="24" alt="" />
+      {dataClass.dataType}
+    </li>
+  ));
+
+  const lastOutputItem = (
+    <li>
+      <Image alt="" src={MoreIcon} />
+      {dataClasses.lowerPriority}
+    </li>
+  );
+
+  return (
+    <>
+      {output} {lastOutputItem}
+    </>
+  );
+}
+
+function makeRecommendationCards(breach: Breach) {
+  const l10n = getL10n();
+  const dataClasses = getSortedDataClasses(breach);
+
+  const priorityRecs = dataClasses.priority.map((dataClass) =>
+    dataClass.recommendations?.map((r) => (
+      <div
+        key={r.ctaHref}
+        className={`breach-detail-recommendation ${r.recIconClassName}`}
+      >
+        <dt>{l10n.getString(r.recommendationCopy.subhead)}</dt>
+        <dd>
+          <p>{l10n.getString(r.recommendationCopy.body)}</p>
+          {r.recommendationCopy.cta ? (
+            <a
+              href={r.ctaHref}
+              target={r.ctaShouldOpenInNewTab ? "_blank" : "_self"}
+            >
+              {l10n.getString(r.recommendationCopy.cta)}
+            </a>
+          ) : null}
+        </dd>
```

Comment:
nit/question: Should there be an encapsulating `<dl>` for these?

Reply:
There absolutely should be. And there was, but it's hard to find with all the indirection going on; I moved it closer in ea5d8dc4c.